test/simulator/misc/random-teams.js

```diff
@@ -26,6 +27,14 @@ function isValidSet(gen, set) {
 	return true;
 }
 
+function validLearnset(move, set, tier) {
+	const validator = TeamValidator(`gen7${tier}`);
+	const template = validator.dex.getTemplate(set.species || set.name);
+	let lsetProblem = validator.checkLearnset(move, template);
+	if (lsetProblem) return false;
```

Comment:
Returning `false` and `true` right beside each other usually points to an issue. Just 
```javascript
return !validator.checkLearnset(move, template);
```
on line 33 and cut out everything after.

Reply:
i have zero clue why i didnt think of that